lib/trino-record-decoder/src/main/java/io/trino/decoder/avro/AvroColumnDecoder.java

```diff
@@ -91,7 +91,7 @@ public AvroColumnDecoder(DecoderColumnHandle columnHandle)
         }
```

Comment:
Why not to add support for short timestamp, date, time and real support to `AvroColumnDecoder`? Then we should be able to drop `PulsarAvroColumnDecoder`. WDYT?

Reply:
The blocker is conversion from shaded version of avro record to original avro record so pulsar connector can utilize `AvroColumnDecoder`, as the whole connector use shaded version of deps.
I tried again and was able to make the conversion this time.
I also move the support of additional type to `AvroColumnDecoder`, but I'm not sure how will this affect existing connectors? My understanding is it shouldn't be problem as it's "additional" suuport.

Reply:
> Why not to add support for short timestamp, date, time and real support to `AvroColumnDecoder`? Then we should be able to drop `PulsarAvroColumnDecoder`. WDYT?

I like this idea. @kokosing is there any existing issue? Or I can try to drive one.

Reply:
FYI https://github.com/trinodb/trino/issues/13069.

BTW pulsar-sql supports Decimal type in:

* https://github.com/apache/pulsar/commit/6b004ed6a2554ab826a00aa2a177963de3c5f44b
* https://github.com/apache/pulsar/commit/0c6e2ca24fd368d11a769233bb2041f6cc4a8374

It's possible to add them to AvroColumnDecoder later.

Also I think that this is not a blocker to add pulsar plugin as we can duplicate code a bit for Pulsar decoders - they have their own logics.